### DISCUSSION TITLE
Does v5 make it strictly required for events to be seriablizable

### DISCUSSION THREAD

**Discussion by aabidsofi19** [statelyai/xstate#4804] [Q&A]
During the migration process from version 4 to version 5, I encountered an issue with passing objects, such as nodes with circular references, in events. This led to a TypeError: Circular value error. 

Surprisingly, this  wasn't mentioned as a breaking change in the migration guide. I'm wondering if this is a bug or if version 5 requires events to strictly serialize objects, which would be understandable for reasons like persistence.

i am not sure if it is a bug or proper behavior , so asking it here .

**Comment by davidkpiano** [comment]
Can you post a code reproduction of the issue?

**Comment by aabidsofi19** [comment]
@davidkpiano  while trying make a minimal  reproduction of the issue , i realized that xstate does still support cyclic events , but the `Cyclic Value error` is thrown when you send a message with a cyclic object  to a actor that has crashed due to some unhandled exception . 

https://codesandbox.io/p/devbox/xstate-cyclic-events-z9592r?file=%2Fsrc%2FApp.tsx
